test/e2e/subnet/subnet_test.go

```diff
@@ -1116,4 +1117,334 @@ var _ = Describe("test subnet", Label("subnet"), func() {
+				_, err = frame.ExecCommandInPod(podList.Items[0].Name, podList.Items[0].Namespace, command, ctx)
+				Expect(err).NotTo(HaveOccurred())
+			}
+			if frame.Info.IpV6Enabled {
+				command := fmt.Sprintf("ip a | grep '%s' -A5 | grep '%s'", common.NIC1, v6IPAddress1)
+				_, err := frame.ExecCommandInPod(podList.Items[0].Name, podList.Items[0].Namespace, command, ctx)
+				Expect(err).NotTo(HaveOccurred())
+
+				command = fmt.Sprintf("ip a | grep '%s' -A5 | grep '%s'", common.NIC2, v6IPAddress2)
+				_, err = frame.ExecCommandInPod(podList.Items[0].Name, podList.Items[0].Namespace, command, ctx)
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			// delete deploy
+			Expect(frame.DeleteDeployment(deployName, namespace)).NotTo(HaveOccurred())
+			ctx, cancel = context.WithTimeout(context.Background(), common.ExecCommandTimeout)
+			defer cancel()
+			if frame.Info.IpV4Enabled {
+				Expect(common.WaitIppoolNumberInSubnet(ctx, frame, v4SubnetName1, 0)).NotTo(HaveOccurred())
+				Expect(common.WaitIppoolNumberInSubnet(ctx, frame, v4SubnetName2, 0)).NotTo(HaveOccurred())
+			}
+			if frame.Info.IpV6Enabled {
+				Expect(common.WaitIppoolNumberInSubnet(ctx, frame, v6SubnetName1, 0)).NotTo(HaveOccurred())
+				Expect(common.WaitIppoolNumberInSubnet(ctx, frame, v6SubnetName2, 0)).NotTo(HaveOccurred())
+			}
+		})
+	})
+
+	Context("The default subnet should support different controller types.", func() {
+		var ClusterDefaultV4SubnetList, ClusterDefaultV6SubnetList []string
+		var v4PoolNameList, v6PoolNameList []string
+		var err error
+		var replicasNum int32 = 1
+		var controllerNum int = 4
+		var controllerNameList []string
+		var podName, deployName, stsName, dsName, jobName string
+
+		BeforeEach(func() {
+			// Get the default subnet
+			// and verify that the default subnet functions properly
+			By("get cluster default subnet")
+			ClusterDefaultV4SubnetList, ClusterDefaultV6SubnetList, err = common.GetClusterDefaultSubnet(frame)
+			Expect(err).NotTo(HaveOccurred())
+			if frame.Info.IpV4Enabled && len(ClusterDefaultV4SubnetList) == 0 {
+				Fail("failed to find cluster ipv4 subnet")
+			}
+			if frame.Info.IpV6Enabled && len(ClusterDefaultV6SubnetList) == 0 {
+				Fail("failed to find cluster ipv6 subnet")
+			}
+
+			By("Generate data for test.")
+			podName = "pod-" + tools.RandomName()
+			controllerNameList = append(controllerNameList, podName)
+			deployName = "deploy-" + tools.RandomName()
+			controllerNameList = append(controllerNameList, deployName)
+			stsName = "sts-" + tools.RandomName()
+			controllerNameList = append(controllerNameList, stsName)
+			dsName = "ds-" + tools.RandomName()
+			controllerNameList = append(controllerNameList, dsName)
+			jobName = "job-" + tools.RandomName()
+			controllerNameList = append(controllerNameList, jobName)
+		})
+
+		It("The default subnet should support different controller types.", Label("I00013", "I00014"), func() {
+			nodeList, err := frame.GetNodeList()
```

Comment:
add comment: this serves for daemonset object to make sure its corresponding pods number